docs/relationship-particles.md

```diff
@@ -0,0 +1,392 @@
+
+#### Boot loader configs
+
+bootc aims to align with the idea of generic bootloader-independent config files where possible; today it uses ostree.  For more on this, see [ostree and bootloaders](https://github.com/ostreedev/ostree/blob/main/docs/bootloaders.md).
+
+### The kernel and initramfs
+
+There is agreement that in order to achieve integrity, there must be a strong link
+between the kernel and the first userspace code that executes in the initial RAM
+disk.
+
+Building on the bootloader statement above: bootc will support [UKI](https://uapi-group.org/specifications/specs/unified_kernel_image/), but not require it.
+
+### The root filesystem
+
+In the bootc model, the root filesystem defaults to a single physical Linux filesystem (e.g. `xfs`, `ext4`, `btrfs` etc.).  It is of course supported to mount other partitions and filesystems; doing so is encouraged even for `/var`.  , where one ends up with some space constraints around the OS `/usr` partition due to dm-verity.
+
+This is a rather large difference already from particles; the root filesystem contains the operating system too; it is not a separate partition.  One thing this helps significantly with is dealing with the "space management" problems that dm-verity introduces (need for
+a partition to have unused empty space to grow, and also a fixed-size ultimate capacity limit).
+
+#### Locating the root
+
+bootc does not mandate or emphasize any particular way to locate the root filesystem;
+parts of the [discoverable partitions specification](https://uapi-group.org/specifications/specs/discoverable_partitions_specification/) specifically the "root partition" may be
+used.  Or, the root filesystem can be found the traditional way, via a local `root=`
+kernel argument.
+
+Another point of contrast from the particle emphasis is that while we encourage encrypting the root filesystem, it is not required.  Particularly some use cases in cloud environments perform encryption at the hypervisor level and do not want additional
+overhead of doing so per virtual machine.
+
+#### Locating the base container image
+
+Until this point, we have been operating under external constraints; no one is creating
+a bootloader that directly understands how to start a container image, for example.
+We've gotten as far as running a Linux userspace in the initial RAM disk, and the
+physical root filesystem is mounted.
+
+Here, we circle back to [composefs](https://github.com/containers/composefs).  One can
+think of composefs as effectively a way to manage something like dm-verity, but using
+files.
+
+What bootc builds on top of that is to target a specfic container image rootfs
+that is part of the "physical" root.  Today, this is implemented again using ostree, via the `ostree=` kernel commandline argument.  In the future, it is likely to be a `bootc.image`.
```

Comment:
> I do try and lean against using kargs these days, in a world where you use a UKI secured thing, there's no difference between using /proc/cmdline and a file in /etc in the initrd. It sounds like we are leaning the same way here.

Yes, this is a good point.  In ostree today we go to some effort actually to avoid changing the initramfs when only userspace changes - and kernel args are really the only way for us to reliably pass state between generic bootloader setups and the initramfs.

But you're right I think in a sealed setup we should indeed not use kargs.